### PR TITLE
server setup

### LINKED ISSUE #13: Server setup
# Project setup
- Create an index.js file in `server/index.js`.
  * import app.js file
  * import the port 
  * set up the listener on the port
- Create an app.js file in `server/app.js`.
  * import router.
  *  add all required middlewares

### PR DESCRIPTION
**Pull request by imansedky5**
fix #13

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const cookieParser = require("cookie-parser");
```
Comment (AdhamHaisami) [inline server/index.js:1]: why import it here and in app.js file?

Comment (imansedky5) [reply server/index.js]: there's no need to call it here, I forgot to remove it

**Comment by AdhamHaisami** [comment]
Good job 👍

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+app.use(middlewares);
+app.use("/api/v1", router);
+
+if (process.env.NODE_ENV === "production") {
+  app.use(express.static(join(__dirname, "..", "client", "build")));
+  app.all("*", (req, res) => {
+    res.sendFile(join(__dirname, "..", "client", "build", "index.html"));
+  });
+}
+
```
Comment (yasmeenattallah) [inline server/app.js:29]: where error handler middleware?

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require("dotenv").config;
+
+const express = require("express");
+const cookieParser = require("cookie-parser");
+const compression = require("compression");
```
Comment (yasmeenattallah) [inline server/app.js:5]: should you add morgan middleware?

File: server/app.js (hunk 2 above)
Comment (imansedky5) [reply server/app.js:37]: I've just added it

**Review by mFlaifel — APPROVED** [review]
good job

**Comment by imansedky5** [comment]
> good job

Thank you!

**Review by yasmeenattallah — APPROVED** [review]

**Review by khamis-hesham — APPROVED** [review]

**Review by AdhamHaisami — APPROVED** [review]

**Comment by AdhamHaisami** [comment]
Fix #49

**Pull request merged**